glusterd2/config.go

```diff
@@ -18,39 +16,48 @@ import (
 	config "github.com/spf13/viper"
 )
 
+const (
+	defaultlogfile = "STDOUT"
+	defaultpeerport = "24008"
+	defaultpeeraddress = ":24008"
+	defaultclientaddress = ":24007"
+	defaultloglevel = "debug"
+)
+
 var (
 	// metrics
-	expConfig = expvar.NewMap("config")
-)
+	expConfig           = expvar.NewMap("config")
 
-const (
-	defaultLogLevel = "debug"
-	defaultConfName = "glusterd2"
+	// Following variables value set by LDFLAGS
+	defaultworkdir = ""
```

Comment:
Is it possible to have just `PREFIX` passed to LDFLAGS and have these derived from that ? That would simplify Makefile too.

For example, in practical use, logdir can default to only `/var/log/glusterfs` or `/usr/local/var/log/glusterfs` where `/usr/local` is the prefix. Similarly, `/var/lib/glusterd2` and `/usr/local/var/lib/glusterd2`

```go
defaultlogdir = prefix + "/var/log/glusterfs"
defaultrundir = prefix + "/var/run/gluster"
```

Reply:
I am not sure about the order of execution when it is set through LDFLAGS. I may have to do those assignments in `init`. I will check and update the PR if works.